src/tigerbeetle.zig

```diff
@@ -78,102 +81,105 @@ pub const Transfer = packed struct {
-    code: u32,
-    flags: CommitFlags,
-    timestamp: u64 = 0,
+pub const CreateAccountResult = enum(u32) {
+    ok,
+    linked_event_failed,
 
-    comptime {
-        assert(@sizeOf(Commit) == 64);
-    }
-};
+    reserved_flag,
+    reserved_field,
 
-pub const CommitFlags = packed struct {
-    linked: bool = false,
-    reject: bool = false,
-    preimage: bool = false,
-    padding: u29 = 0,
+    id_must_not_be_zero,
+    ledger_must_not_be_zero,
+    code_must_not_be_zero,
 
-    comptime {
-        assert(@sizeOf(CommitFlags) == @sizeOf(u32));
-    }
-};
+    mutually_exclusive_flags,
 
-pub const CreateAccountResult = enum(u32) {
-    ok,
-    linked_event_failed,
-    exists,
-    exists_with_different_user_data,
-    exists_with_different_reserved_field,
-    exists_with_different_unit,
-    exists_with_different_code,
-    exists_with_different_flags,
     exceeds_credits,
     exceeds_debits,
-    reserved_field,
-    reserved_flag_padding,
-};
 
-pub const CreateTransferResult = enum(u32) {
-    ok,
-    linked_event_failed,
-    exists,
-    exists_with_different_debit_account_id,
-    exists_with_different_credit_account_id,
     exists_with_different_user_data,
-    exists_with_different_reserved_field,
+    exists_with_different_ledger,
     exists_with_different_code,
-    exists_with_different_amount,
-    exists_with_different_timeout,
     exists_with_different_flags,
-    exists_and_already_committed_and_accepted,
-    exists_and_already_committed_and_rejected,
-    reserved_field,
-    reserved_flag_padding,
-    debit_account_not_found,
-    credit_account_not_found,
-    accounts_are_the_same,
-    accounts_have_different_units,
-    amount_is_zero,
-    exceeds_credits,
-    exceeds_debits,
-    two_phase_commit_must_timeout,
-    timeout_reserved_for_two_phase_commit,
+    exists,
 };
 
-pub const CommitTransferResult = enum(u32) {
+pub const CreateTransferResult = enum(u32) {
     ok,
     linked_event_failed,
+
+    reserved_flag,
     reserved_field,
-    reserved_flag_padding,
-    transfer_not_found,
-    transfer_not_two_phase_commit,
-    transfer_expired,
-    already_committed,
-    already_committed_but_accepted,
-    already_committed_but_rejected,
+
+    id_must_not_be_zero,
+    debit_account_id_must_not_be_zero,
+    credit_account_id_must_not_be_zero,
+    accounts_must_be_different,
+
+    pending_id_must_be_zero,
+    pending_transfer_must_timeout,
+    timeout_reserved_for_pending_transfer,
+
+    ledger_must_not_be_zero,
+    code_must_not_be_zero,
+    amount_must_not_be_zero,
+
     debit_account_not_found,
     credit_account_not_found,
-    debit_amount_was_not_reserved,
-    credit_amount_was_not_reserved,
+
+    accounts_must_have_the_same_ledger,
+    transfer_must_have_the_same_ledger_as_accounts,
+
+    overflows_debits,
+    overflows_debits_pending,
+    overflows_debits_posted,
+    overflows_credits,
+    overflows_credits_pending,
+    overflows_credits_posted,
```

Comment:
These should be reordered back to how they were to match the order that they are returned from the state machine.

Reply:
Fixed.